### PR TITLE
Updating SmartSim version for release

### PR DESCRIPTION
**Pull request by amandarichardsonn**
This PR changes most instances of 0.6.0 to 0.6.1

### CODE REVIEW

File: .github/workflows/build_docs.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ jobs:
           git config --local user.name "GitHub Action"
           git commit -m "Update develop documentation"
 
-      - uses: ad-m/github-push-action@v0.6.0
+      - uses: ad-m/github-push-action@v0.6.1
```
Comment (MattToast) [inline .github/workflows/build_docs.yml:68]: I'm guessing this action got caught in then "find/replace" crossfire? Should be free to revert!

File: doc/_static/version_names.json [hunk 2]
```diff
@@ -12,6 +13,8 @@
     "version_urls": [
         "https://www.craylabs.org/develop/overview.html",
         "https://www.craylabs.org/docs/overview.html",
+        "https://www.craylabs.org/docs/versions/0.6.1/overview.html",
```
Comment (MattToast) [inline doc/_static/version_names.json:16]: Could you drop this link? The docs for `0.6.1` should be (eventually, there is a little manual intervention that needs to happen) at `https://www.craylabs.org/docs/overview.html`

Comment (MattToast) [reply doc/_static/version_names.json:16]: i.e. the `"version_names"` list and then `"version_urls"` list should be of the same length, where index `i` in `"version_names"` maps to index `i` in `"version_urls"`.

Why it was decided to make these as two discrete lists when we are very clearly mapping keys to values is anyone's guess, lol ¯\\\_(ツ)\_/¯

**Review by MattToast — CHANGES_REQUESTED** [review]
Two VERY small changes, but otherwise LGTM!!

File: doc/_static/version_names.json (hunk 2 above)
Comment (amandarichardsonn) [reply doc/_static/version_names.json:16]: merppp thank u for catchin! I see

File: doc/_static/version_names.json [hunk 3]
```diff
@@ -1,7 +1,8 @@
 {
     "version_names":[
         "develop (unstable)",
-        "0.6.0 (stable)",
+        "0.6.1",
```
Comment (MattToast) [inline doc/_static/version_names.json:4]: Can we re-add the `(stable)` tag here?

**Review by MattToast — APPROVED** [review]
One last small ask, but nothing worth holding up approval over. LGTM!!

File: .wci.yml [hunk 4]
```diff
@@ -22,7 +22,7 @@
     language: Python
 
     release:
-      version: 0.6.0
+      version: 0.6.1
       date: 2023-12-18
```
Comment (al-rigazzi) [inline .wci.yml:26]: This date needs to be updated

**Review by al-rigazzi — CHANGES_REQUESTED** [review]
Please update release date on .wci.yml

File: .wci.yml (hunk 4 above)
Comment (al-rigazzi) [reply .wci.yml]: Done

Comment (al-rigazzi) [reply .wci.yml]: Thanks

Comment (al-rigazzi) [reply .wci.yml]: Bet!

**Review by al-rigazzi — APPROVED** [review]
LGTM, thanks for fixing that bad date issue.

**Pull request merged**